pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/dictionary/DictionaryIndexType.java

```diff
@@ -236,10 +237,18 @@ public static boolean ignoreDictionaryOverride(boolean optimizeDictionary,
       // Do not create dictionary if index size with dictionary is going to be larger than index size without dictionary
       // This is done to reduce the cost of dictionary for high cardinality columns
       // Off by default and needs optimizeDictionary to be set to true
-      if (fieldSpec.isSingleValueField() && fieldSpec.getDataType().isFixedWidth()) {
-        // if you can safely enable dictionary, you can ignore overrides
-        return canSafelyCreateDictionaryWithinThreshold(cardinality, totalNumberOfEntries,
-            noDictionarySizeRatioThreshold, fieldSpec);
+      if (fieldSpec.isSingleValueField()) {
+        if (fieldSpec.getDataType().isFixedWidth()) {
+          // if you can safely enable dictionary, you can ignore overrides
+          return canSafelyCreateDictionaryWithinThreshold(cardinality, totalNumberOfEntries,
```

Comment:
For fixed width type `noDictionarySizeRatioThreshold` is always picked even if it is not configured (default value being used). Do we want to provide a way so that `noDictionaryCardinalityRatioThreshold` is picked when `noDictionarySizeRatioThreshold` is not configured or explicitly disabled?

Reply:
I was thinking the original `noDictionarySizeRatioThreshold` config should be a better overall heuristic. It also keeps the user flow simple:
- optimize fixed width type col? -> use `noDictionarySizeRatioThreshold`
- optimize var width type col? -> use `noDictionaryCardinalityRatioThreshold`